src/databricks/sql/auth/endpoint.py

```diff
@@ -0,0 +1,107 @@
+#
+# It implements all the cloud specific OAuth configuration/metadata
+#
+#   Azure: It uses AAD
+#   AWS: It uses Databricks internal IdP
+#   GCP: Not support yet
+#
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional, List
+import os
+
+OIDC_REDIRECTOR_PATH = "oidc"
+
+
+class OAuthScope:
+    OFFLINE_ACCESS = "offline_access"
+    SQL = "sql"
+
+
+class CloudType(Enum):
+    AWS = "aws"
+    AZURE = "azure"
+    GCP = "gcp"
+
+
+# Infer cloud type from Databricks SQL instance hostname
+def infer_cloud_from_host(hostname: str) -> Optional[CloudType]:
+    # normalize
+    host = hostname.lower().replace("https://", "").split("/")[0]
+
+    if host.endswith(".azuredatabricks.net"):
+        return CloudType.AZURE
+    elif host.endswith(".gcp.databricks.com"):
+        return CloudType.GCP
+    elif host.endswith("cloud.databricks.com"):
+        return CloudType.AWS
+    else:
+        return None
+
+
+def get_databricks_oidc_url(hostname: str):
+    maybe_scheme = "https://" if not hostname.startswith("https://") else ""
+    maybe_trailing_slash = "/" if not hostname.endswith("/") else ""
+    return f"{maybe_scheme}{hostname}{maybe_trailing_slash}{OIDC_REDIRECTOR_PATH}"
+
+
+class OAuthEndpoints(ABC):
```

Comment:
It's unusual to me to see Python class names expressed as plurals. If this object exposes a collection then that should be demonstrated by the name. Something like `OAuthEndpointCollection` or `OAuthEndpoingList` (note that both are singular).

Reply:
I will rename it. Thanks.